File: MorWalPizVideo.Models/Models/QuickLinks.cs
using MongoDB.Bson.Serialization.Attributes;
using System.Runtime.Serialization;

namespace MorWalPizVideo.Server.Models;

[DataContract]
public enum QuickLinkKind
{
    [EnumMember] External = 0,
    [EnumMember] Telegram = 1,
    [EnumMember] Instagram = 2,
    [EnumMember] Facebook = 3,
    [EnumMember] Video = 4
}

[BsonIgnoreExtraElements]
[DataContract]
public sealed record QuickLink(
    [property: DataMember][property: BsonElement("kind")] QuickLinkKind Kind,
    [property: DataMember][property: BsonElement("targetUrl")] string TargetUrl,
    [property: DataMember][property: BsonElement("title")] string? Title = null,
    [property: DataMember][property: BsonElement("subtitle")] string? Subtitle = null,
    [property: DataMember][property: BsonElement("label")] string? Label = null,
    [property: DataMember][property: BsonElement("imageUrl")] string? ImageUrl = null,
    [property: DataMember][property: BsonElement("icon")] string? Icon = null,
    [property: DataMember][property: BsonElement("provider")] string? Provider = null);

[BsonIgnoreExtraElements]
[DataContract]
public sealed record QuickLinks(
    [property: DataMember][property: BsonElement("title")] string Title,
    [property: DataMember][property: BsonElement("subtitle")] string? Subtitle,
    [property: DataMember][property: BsonElement("url")] string Url,
    [property: DataMember][property: BsonElement("links")] QuickLink[] Links) : BaseEntity
{
    [DataMember]
    [BsonElement("channelId")]
    public string ChannelId { get; init; } = string.Empty;

    public static string NormalizeUrl(string? url) => (url ?? string.Empty).Trim().Trim('/').ToLowerInvariant();
}

File: frontend/fe-packages/models/src/quickLinks.ts
export const QuickLinkKind = {
  External: 0,
  Telegram: 1,
  Instagram: 2,
  Facebook: 3,
  Video: 4,
} as const;

export type QuickLinkKind = typeof QuickLinkKind[keyof typeof QuickLinkKind];

export interface QuickLink {
  kind: QuickLinkKind;
  targetUrl: string;
  title?: string;
  subtitle?: string;
  label?: string;
  imageUrl?: string;
  icon?: string;
  provider?: string;
}

export interface QuickLinks {
  id?: string;
  channelId?: string;
  title: string;
  subtitle?: string;
  url: string;
  links: QuickLink[];
  creationDateTime?: string;
}

export interface CreateQuickLinksDTO {
  channelId?: string;
  title: string;
  subtitle?: string;
  url: string;
  links: QuickLink[];
}

export type UpdateQuickLinksDTO = CreateQuickLinksDTO;
